web/ui/src/features/river-js/RiverValue.tsx

```diff
@@ -1,30 +1,56 @@
-import React, { FC, Fragment, ReactElement } from 'react';
+import React, { FC, Fragment } from 'react';
 
 import { ObjectField, Value, ValueType } from './types';
 
 import styles from './RiverValue.module.css';
 
 export interface RiverValueProps {
   value: Value;
+  name?: string;
+  nthChild?: number;
 }
 
+// TO track the length of the object key value pair
+const DIVISION_LENGTH = 20;
+
 /**
  * RiverValue emits a paragraph which represents a River value.
  */
-export const RiverValue: FC<RiverValueProps> = (props) => {
+export const RiverValue: FC<RiverValueProps> = ({ value, name, nthChild }) => {
   return (
-    <p className={styles.value}>
-      <ValueRenderer value={props.value} indentLevel={0} />
-    </p>
+    <div className={styles.value}>
+      <ValueRenderer value={value} nthChild={nthChild} name={name} />
+    </div>
   );
 };
 
-type valueRendererProps = RiverValueProps & {
-  indentLevel: number;
-};
+const ValueRenderer: FC<RiverValueProps> = ({ value, name, nthChild }) => {
+  const backgroundColor = nthChild && nthChild % 2 === 1 ? '#f4f5f5' : 'white';
+
+  /**
+   * Renderer for the river format for target and label object
+   */
+  const renderGrid = (partition: ObjectField[]) => {
+    // const gridTemplateColumns = '10% 1% 69%';
+
+    let maxLength = 0;
 
-const ValueRenderer: FC<valueRendererProps> = (props) => {
-  const value = props.value;
+    partition.forEach(({ key }) => {
+      return key.length > maxLength ? (maxLength = key.length) : maxLength;
+    });
+
+    const gridTemplateColumns = maxLength < DIVISION_LENGTH ? '10% 1% 89%' : undefined;
```

Comment:
Set a width on grid column depending on division length. (This way the `=` symbol will be lined up otherwise there will be a big gap in the shorter key group such as `labels`). Comparison ss below with dynamic/static grid width:
 
![Screen Shot 2023-01-23 at 11 59 36 AM](https://user-images.githubusercontent.com/112968312/214137880-2e0ea3f5-1319-4819-ba44-169eb542a80f.png)
 
 
![Screen Shot 2023-01-23 at 11 59 58 AM](https://user-images.githubusercontent.com/112968312/214137955-0aff33f1-4fc2-4eed-9f3b-3a2006715aa0.png)

Due to the reuse of `<table>`, and the diversity of the key:value pair. We need to adjust the width in order to fit both the longer keys and shorter keys.